### PR TITLE
[FLINK-4602][State] Move RocksDB statebackend classes to o.a.f.state.rocksdb package.

### PR DESCRIPTION
**Pull request by AlexYinHan**
## What is the purpose of the change

This PR moves RocksDB statebackend classes to o.a.f.state.rocksdb package, so as to close [FLINK-4602](https://issues.apache.org/jira/browse/FLINK-4602) and [FLIP-349](https://cwiki.apache.org/confluence/display/FLINK/FLIP-349%3A+Move+RocksDB+statebackend+classes+to+o.a.f.state.rocksdb+package).

## Brief change log

Move rocksdb classes from ```org.apache.flink.contrib.streaming.state``` to ```org.apache.flink.state.rocksdb```.

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Comment by AlexYinHan** [comment]
@flinkbot run azure

**Review by Zakelly — CHANGES_REQUESTED** [review]
@AlexYinHan Thanks for the effort! Only one thing: I think we should avoid introducing breaking change by keeping the `@PublicEvolving` APIs but marked `@Deprecated`. There better be a `EmbeddedRocksDBStateBackend` and `EmbeddedRocksDBStateBackendFactory` under the original package name, which simply extends the moved corresponding classes.

**Comment by AlexYinHan** [comment]
@flinkbot run azure

File: flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/EmbeddedRocksDBStateBackendFactory.java [hunk 1]
```diff
@@ -1,35 +1,10 @@
...
+@Deprecated
 public class EmbeddedRocksDBStateBackendFactory
-        implements StateBackendFactory<EmbeddedRocksDBStateBackend> {
-    @Override
-    public EmbeddedRocksDBStateBackend createFromConfig(
-            ReadableConfig config, ClassLoader classLoader) throws IllegalConfigurationException {
-        return new EmbeddedRocksDBStateBackend().configure(config, classLoader);
-    }
-}
+        extends org.apache.flink.state.rocksdb.EmbeddedRocksDBStateBackendFactory {}
```
Comment (Zakelly) [inline flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/EmbeddedRocksDBStateBackendFactory.java:10]: IIUC, this still changes the signature. Maybe implement the #createFromConfig to create new `EmbeddedRocksDBStateBackend`?

**Comment by AlexYinHan** [comment]
@Zakelly I have also processed the remaining 5 classes annotated with PublicEvolving, and added the license to all the new files. I also executed the command to regenerate the documentation, and the doc files remained unchanged before and after the command execution. PTAL. 

```
ConfigurableRocksDBOptionsFactory.java
RocksDBConfigurableOptions.java
RocksDBNativeMetricOptions.java
RocksDBOptions.java
RocksDBOptionsFactory.java
```

**Comment by Zakelly** [comment]
@AlexYinHan  Thanks for the update! I think you should keep the `@PublicEvolving` for those old classes.

**Review by Zakelly — APPROVED** [review]
LGTM

**Pull request merged**